packages/app-desktop/gui/NoteEditor/utils/contextMenu.ts

```diff
@@ -100,33 +77,49 @@ export function menuItems(dispatch: Function): ContextMenuItems {
 			onAction: async (options: ContextMenuOptions) => {
 				await openItemById(options.resourceId, dispatch);
 			},
-			isActive: (itemType: ContextMenuItemType) => itemType === ContextMenuItemType.Image || itemType === ContextMenuItemType.Resource,
+			isActive: (itemType: ContextMenuItemType, options: ContextMenuOptions) => !options.isTemp && itemType === ContextMenuItemType.Image || itemType === ContextMenuItemType.Resource,
 		},
 		saveAs: {
 			label: _('Save as...'),
 			onAction: async (options: ContextMenuOptions) => {
-				const { resourcePath, resource } = await resourceInfo(options);
-				const filePath = await bridge().showSaveDialog({
-					defaultPath: resource.filename ? resource.filename : resource.title,
-				});
-				if (!filePath) return;
-				await fs.copy(resourcePath, filePath);
+				await saveResource(options);
```

Comment:
Restore the above code, and create a new function to save the message dynamic content to file. I think you're trying to cram the feature into existing unrelated code.